tutorpicasso/commands/run_extra_commands.py

```
@@ -0,0 +1,157 @@
+    """
+    splitted_commands = [
+        split_string(command, COMMAND_CHAINING_OPERATORS) for command in commands
+    ]
+    flat_commands_list: chain[str] = chain.from_iterable(splitted_commands)
+
+    invalid_commands = []
+    misspelled_commands = []
+    for command in flat_commands_list:
+        if "tutor" not in command.lower():
+            if find_tutor_misspelled(command):
+                misspelled_commands.append(command)
+            else:
+                invalid_commands.append(command)
+
+        error_message = ""
+
+        if invalid_commands:
+            error_message += (
+                f"Found some issues with the commands:\n\n"
+                f"=> Invalid commands: {', '.join(invalid_commands)}\n"
+            )
+
+        if misspelled_commands:
+            error_message += (
+                f"=> Misspelled commands: {', '.join(misspelled_commands)}\n"
+            )
+
+        if error_message:
+            error_message += (
+                "Take a look at the official Tutor commands: "
+                "https://docs.tutor.edly.io/reference/cli/index.html"
+            )
+            return error_message
+    return ""
+
+
+def run_command(command: str) -> None:
+    """
+    Run an extra command.
+
+    This method runs the extra command provided.
+
+    Args:
+        command (str): Tutor command.
+    """
+    try:
+        with subprocess.Popen(
+            command,
+            shell=True,
+            executable="/bin/bash",
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        ) as process:
+
+            stdout, stderr = process.communicate()
+
+            if process.returncode != 0 or "error" in stderr.lower():
+                raise subprocess.CalledProcessError(
+                    process.returncode, command, output=stdout, stderr=stderr
+                )
```

Could we raise the click exception here instead? We're catching the exception in 107L and re-raising it there without any other step in between. Or does the exception in line 107L have another purpose?

---

It has not. I will use your suggestion.